staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured.go

```diff
@@ -82,9 +82,9 @@ func (obj *Unstructured) EachListItem(fn func(runtime.Object) error) error {
 
 func (obj *Unstructured) UnstructuredContent() map[string]interface{} {
 	if obj.Object == nil {
-		obj.Object = make(map[string]interface{})
+		return make(map[string]interface{})
 	}
-	return obj.Object
+	return runtime.DeepCopyJSON(obj.Object)
```

Comment:
I don't think we should do this. We return shallow copies in many locations.  Callers shouldn't modify them.  The cost of doing the copies exceeds the cost of being careful.

Reply:
@deads2k I'm ok to get rid of this change, but an in-memory copy shouldn't be extremely expensive right?

Reply:
@rchakra3 A deep-copy of non-trivial types leads to many memory allocations. So it is far from being for free. If we change this, we need load testing results beforehand.